qchem/CHANGELOG.md

```diff
@@ -2,6 +2,21 @@
 
 <h3>New features since last release</h3>
 
+* The ``derivative()`` and ``gradient()`` functions have been added to compute
```

Comment:
Is it necessary to have two functions? If ``derivative()`` is not used for other purposes, it might be merged with ``gradient()``.

Reply:
Thanks for rising this point. 
- The ``derivative()`` function is needed to compute the Hessian of the total energy ``E(x)``. 
![Screenshot from 2021-02-23 09-32-00](https://user-images.githubusercontent.com/45397799/108858613-5d066880-75ba-11eb-95e3-10fbf3d78b06.png)

- Providing a gradient function might be gradient function  convenient to evaluate the gradient of ``E(x)`` which we need for the geometry optimization.

```python
# gradient E(x)
def gradient(x, params):
    nabla = [qml.ExpvalCost(circuit, grad, dev)(params) for grad in qml.qchem.gradient(H, x)]
    return np.array([nabla])
```
The latter can also be done straightforwardly as follows:

```python
# gradient E(x)
def gradient(x, params):
    nabla = [qml.ExpvalCost(circuit, qml.qchem.derivative(H, x, i), dev)(params) for i in len(x)]
    return np.array([nabla])
```
thoughts here?. do we want to have the gradient function?.

Reply:
I vote for not having the `gradient` function since `derivative` is able to do the same job using a simple loop (based on the two examples provided above).

Reply:
We definitely need the `derivative` function. It's true that computing the gradient is then a simple loop, but the upside is that it makes the UI more elegant since the gradient is the central object that appears in the optimization. 

I prefer to keep both since there isn't much downside to having `gradient`, but I'm open to being convinced otherwise